Automattic/jetpack
# Update use-module-status hook to use modules_store

Fixes https://github.com/Automattic/jetpack/issues/34588

## Proposed changes:

In order to have a consistent state, remove unnecessary loading and avoid unnecessary requests we are utilizing newly create `modules_store` within same package. This should also solve issues when activating block that is present multiple times on the page.
Tests are updated as well. 
Coverage for Redux store itself in this PR: https://github.com/Automattic/jetpack/pull/34835

**Before**

https://github.com/Automattic/jetpack/assets/60262784/f58a1597-414e-4687-91dc-11f27085e77c

**After**

https://github.com/Automattic/jetpack/assets/60262784/ef1ea817-a550-4068-a8f8-9da90a6f0b2d

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:





Generally, all features using `use-module-status` should be tested (mostly blocks relying on modules). These include: `forms`, `related-posts`, `subscriptions`, `top-posts`, `wordads`, `likes` and `sharing` (modules), `seo-tools`. 

- Spin up your site with local changes
- Navigate to `wp-admin/admin.php?page=jetpack_modules` and disable needed modules
- Build packages,  at least these: `jetpack build packages/forms`, `jetpack build plugins/jetpack`, `jetpack build packages/videopress`
- Create a new post and add blocks you are testing. Activate module. Observe the behavior.
- Check features in Jetpack sidebar of the post.
- Verify self-hosted and Simple/Atomic work normally.
